pkg/cri/resource-manager/control/blockio/blockio.go

```diff
@@ -134,26 +135,48 @@ func (ctl *blockioctl) BlockIOClass(c cache.Container) string {
 		}
 	}
 
-	log.Debug("BlockIO class for %s (%s): %q", c.PrettyName(), cclass, blockioclass)
-
 	return blockioclass
 }
 
-// configNotify is blockio class mapping configuration notification callback.
+// configNotify is blockio class mapping and class definition configuration callback
 func (ctl *blockioctl) configNotify(event config.Event, source config.Source) error {
-	// BlockIO class mapping in opt (in flags.go) has changed.
-	// It will affect only new pods. cgroupsblkio parameters of running pods
-	// is not changed.
-	log.Info("class mapping configuration updated")
+	ignoreErrors := (event == config.RevertEvent)
+	err := blockio.UpdateOciConfig(ignoreErrors)
+	if err != nil && !ignoreErrors {
+		return err
+	}
+	// Possible errors in reconfiguring running containers are not errors in
+	// the updated configuration, therefore silently ignored.
+	ctl.reconfigureRunningContainers()
 	return nil
 }
 
+// reconfigureRunningContainers force setting current blockio configuration to all containers running on the node
+func (ctl *blockioctl) reconfigureRunningContainers() error {
+	var errors *multierror.Error
+	if ctl.cache == nil {
+		return nil
+	}
+	for _, pod := range ctl.cache.GetPods() {
+		for _, c := range pod.GetContainers() {
```

Comment:
Can you switch this loop to use `cache.GetContainers()`. As you pointed out `pod.GetContainers()` fails to properly filter duplicates from the returned slice. I'll fix that but `cache.GetContainers()` does not have the same problem and since you don't do anything with pods it looks more appropriate anyway for this case...

Reply:
Yeah... after having a look inside `pod.GetContainers`, it is definitely better to iterate directly on cache level. Fixed.